File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBFloat.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

/**
 * <code>PBFloat</code> is the representation of Prolog integers.
 */

class PBFloat extends PBAtomic
{
    double doubleValue;

    /**
     * Creates a new <code>PBFloat</code> instance with the specified value.
     */
    PBFloat(double value)
    {
        super(Double.toString(value));
        doubleValue = value;
    }

    PBFloat(double value, String name)
    {
        super(name);
        doubleValue = value;
    }

    int getType()
    {
        return FLOAT;
    }

    public boolean isFloat()
    {
        return true;
    }

    public double floatValue()
    {
//      return Double.parseDouble(name);
        return doubleValue;
    }

    String toPrologString()
    {
        return toString();
    }

    public String toString()
    {
        return Double.toString(doubleValue);
    }

}

File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBFloat.cs

using System;

namespace se.sics.prologbeans
{
	
	/// <summary> <code>PBFloat</code> is the representation of Prolog integers.</summary>
	
	class PBFloat:PBAtomic
	{
		override sealed internal int Type {
			get {
				return FLOAT;
			}
			
		}

		override sealed public bool Float {
			get {
				return true;
			}
			
		}

		override sealed public bool Number {
			get {
				return true;
			}
		}

		internal double doubleValue;

		/// <summary> Creates a new <code>PBFloat</code> instance with the specified value.</summary>
		internal PBFloat (double value) : base (value.ToString ())
		{
			doubleValue = value;
		}

		internal PBFloat (double value, string name) : base (name)
		{
			doubleValue = value;
		}

		public override double floatValue ()
		{
			//      return Double.parseDouble(name);
			return doubleValue;
		}

		internal override string toPrologString ()
		{
			return ToString ();
		}

		public override string ToString ()
		{
			return doubleValue.ToString ();
		}
	}
}